services/hanke-service/src/integrationTest/kotlin/fi/hel/haitaton/hanke/attachment/application/ApplicationAttachmentMigratorITest.kt

```diff
@@ -0,0 +1,146 @@
+            assertThat(attachmentRepository.findAll()).hasSize(1)
+            assertThat(contentRepository.findAll()).hasSize(1)
+            val content = contentRepository.findByIdOrNull(attachment.id!!)!!
+            assertThat(fileClient.listBlobs(Container.HAKEMUS_LIITTEET)).isEmpty()
+
+            val attachmentWithContent = attachmentMigrator.findAttachmentWithDatabaseContent()
+
+            assertThat(attachmentWithContent).isNotNull().all {
+                prop(ApplicationAttachmentWithContent::id).isEqualTo(attachment.id!!)
+                prop(ApplicationAttachmentWithContent::applicationId)
+                    .isEqualTo(attachment.applicationId)
+                prop(ApplicationAttachmentWithContent::content).all {
+                    prop(AttachmentContent::fileName).isEqualTo(attachment.fileName)
+                    prop(AttachmentContent::contentType).isEqualTo(attachment.contentType)
+                    prop(AttachmentContent::bytes).isEqualTo(content.content)
+                }
+            }
+        }
+    }
+
+    @Nested
+    inner class Migrate {
+        @Test
+        fun `Should migrate content to blob`() {
+            attachmentFactory.save().withDbContent().value
+            assertThat(attachmentRepository.findAll()).hasSize(1)
+            assertThat(contentRepository.findAll()).hasSize(1)
+            assertThat(fileClient.listBlobs(Container.HAKEMUS_LIITTEET)).isEmpty()
+            val attachmentWithContent = attachmentMigrator.findAttachmentWithDatabaseContent()!!
+
+            val blobPath = attachmentMigrator.migrate(attachmentWithContent)
+
+            assertThat(attachmentRepository.findAll()).hasSize(1)
+            assertThat(contentRepository.findAll()).hasSize(1)
+            assertThat(fileClient.listBlobs(Container.HAKEMUS_LIITTEET)).hasSize(1)
+            assertThat(fileClient.listBlobs(Container.HAKEMUS_LIITTEET).first()).all {
+                prop(TestFile::path).isEqualTo(blobPath)
+            }
+        }
+    }
+
+    @Nested
+    @Transactional
+    inner class SetBlobPathAndCleanup {
+        @Test
+        fun `Should throw AttachmentNotFoundException if attachment is not in db`() {
+            val attachment = attachmentFactory.save().withDbContent().value
+            val blobPath = generateBlobPath(attachment.applicationId)
+            attachmentRepository.delete(attachment)
+            assertThat(attachmentRepository.findAll()).isEmpty()
+
+            assertThrows<AttachmentNotFoundException> {
+                attachmentMigrator.setBlobPathAndCleanup(attachment.id!!, blobPath)
+            }
+
+            assertThat(attachmentRepository.findAll()).isEmpty()
+        }
+
+        @Test
+        fun `Should set blob path and delete content from db`() {
+            val attachment = attachmentFactory.save().withDbContent().value
+            val blobPath = generateBlobPath(attachment.applicationId)
```

Comment:
I think it would be better to be explicit with blobPath here. So we know `setBlobPathAndCleanup` uses the parameter and doesn't compute it independently.
```suggestion
            val blobPath = "some/path"
```

Reply:
Fixed.